tests/math/sequence/test_sequential_funcs.py

```diff
@@ -46,6 +46,31 @@ def ddof_arg(*arrs):
     return st.integers(0, min_side - 1) if min_side else st.just(0)
 
 
+@st.composite
+def gen_average_args(draw, arr):
+    # Weight is either
+    # - the same shape as arr OR
+    # - 1D with compatible length
+    # See: https://github.com/numpy/numpy/blob/c31cc36a8a814ed4844a2a553454185601914a5a/numpy/lib/function_base.py#L397
+    wt_same_shape, wt_1D = range(2)
+    if arr.ndim == 0 or draw(st.one_of(st.just(wt_same_shape), st.just(wt_1D))) == wt_same_shape:
+        axis = draw(axis_arg(arr))
+        wt_shape = arr.shape
+    else: # wt_1D
+        # Only integer axis is supported for 1D weights
+        axis = draw(st.integers(min_value=0, max_value=len(arr.shape) - 1))
```

Comment:
Negative axes are permitted:

```suggestion
        axis = draw(st.integers(min_value=-arr.ndim, max_value=arr.ndim - 1))
```